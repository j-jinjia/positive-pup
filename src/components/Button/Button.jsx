import "./Button.scss";

const Button = (props) => {
  const { buttonText, isSecondary, isSquare, link, isSmallText } = props;
  let buttonClassName = "button";
  buttonClassName += isSecondary ? " button--secondary" : " button--primary";

  if (link) buttonClassName += " button--link";
  if (isSquare) buttonClassName += " button--squared";
  if (isSmallText) buttonClassName += " button--small-text";

  return link ? (
    <div className={buttonClassName}>{link}</div>
  ) : (
    <button className={buttonClassName}>{buttonText}</button>
  );
};

export default Button;
