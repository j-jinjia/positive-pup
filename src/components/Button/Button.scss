@use "../../styles/variables" as *;

.button {
  padding: 15px 38px;
  border: none;
  border-radius: 8px;

  &--squared {
    border-radius: 0;
  }

  &--primary {
    color: $colour-secondary;
    background-color: $colour-primary;
    font-weight: bold;
  }

  &--secondary {
    color: $colour-primary;
    background-color: transparent;
    border: 2px solid $colour-primary;
  }

  &:hover {
    cursor: pointer;
  }
}
