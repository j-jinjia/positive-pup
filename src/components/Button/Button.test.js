import { render } from "@testing-library/react";
import Button from "./Button";

it("should match button component snapshot", () => {
  const { container } = render(<Button />);

  expect(container).toMatchSnapshot();
});
