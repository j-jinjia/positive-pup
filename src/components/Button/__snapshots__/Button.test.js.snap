// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match button component snapshot 1`] = `
<div>
  <button
    class="button button--primary"
  />
</div>
`;
